pkg/k8s/synced/crd.go

```diff
@@ -41,21 +41,16 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
-func (k *K8sWatcher) crdsInit(ctx context.Context, watcherClient apiextclientset.Interface) error {
-	crds := newCRDState(
-		map[string]bool{
-			cnpCRD:  false,
-			ccnpCRD: false,
-			cepCRD:  false,
-			cnCRD:   false,
-			cidCRD:  false,
-			clrpCRD: false,
-		},
-	)
+const (
+	k8sAPIGroupCRD = "CustomResourceDefinition"
+)
+
+func WaitForCRDsToRegister(ctx context.Context, rs *Resources, ag *APIGroups) error {
```

Comment:
Maybe we can consider renaming this function to `SyncCRDs` and having the following godoc:

```
SyncCRDs will sync Cilium CRDs to ensure that they have all been installed inside the K8s cluster. These CRDs are added by the Cilium Operator. This function will block until it finds all the CRDs or if a timeout occurs.

Reply:
Done, thanks!